iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant_internal.hpp

```diff
@@ -109,7 +109,7 @@ struct call_at_index
         }
     }
 
-    static void move(const uint64_t index, byte_t* source, byte_t* destination) noexcept
+    static void move(const uint64_t index, void* source, void* destination) noexcept
```

Comment:
Oh boy, I hate to do this but
```suggestion
    static void move(const uint64_t index, const void* const source, void* const destination) noexcept
```
:sweat_smile:

Reply:
@elBoberido I didn't follow your approach completely for move so that it doesn't degrade to a copy operation. Hope I didn't mess up the const correctness - it is late...

Reply:
@FerdinandSpitzschnueffler yeah, I mixed it up :sweat_smile: